docs/dev/design/redirects.rst

```diff
@@ -0,0 +1,338 @@
+  They also allow you to set the status code of the redirect,
+  and redirects can be specific in a ``_redirects`` file.
+
+  - Forced redirects
+  - Match query arguments
+  - Match by country/language and cookies
+  - Per-domain and protocol redirects
+  - In case of multiple matches, the topmost redirect will be used.
+
+  https://docs.netlify.com/routing/redirects/
+
+Improvements
+------------
+
+General improvements
+~~~~~~~~~~~~~~~~~~~~
+
+The following improvements will be applied to all types of redirects.
+
+- Allow choosing the status code of the redirect.
+  We already have a field for this, but it's not exposed to users.
+- Allow to explicitly define the order of redirects.
+  This will be similar to the automation rules feature,
+  where users can reorder the rules so the most specific ones are first.
+  We currently rely on the implicit order of the redirects (updated_at).
+- Allow to disable redirects.
+  It's useful when testing redirects, or when debugging a problem.
+  Instead of having to re-create the redirect,
+  we can just disable it and re-enable it later.
+- Allow to add a short description.
+  It's useful to document why the redirect was created.
+
+Don't run redirects on domains from pull request previews
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We currently run redirects on domains from pull request previews,
+this is a problem when moving a whole project to a new domain.
+
+We don't the need to run redirects on external domains, they
+should be treated as temporary domains.
+
+Normalize paths with trailing slashes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, if users want to redirect a path with a trailing slash and without it,
+they need to create two separate redirects (``/page/`` and ``/page``).
+
+We can simplify this by normalizing the path before matching it.
+
+For example:
+
+From:
+  ``/page/``
+To:
+  ``/new/page``
+
+The from path will be normalized to ``/page``,
+and the filename to match will also be normalized before matching it.
+This is similar to what Netlify does:
+https://docs.netlify.com/routing/redirects/redirect-options/#trailing-slash.
+
+Page and exact redirects without a wildcard at the end will be normalized,
+all other redirects need to be matched as is.
+
+This makes it impossible to match a path with a trailing slash.
+
+Explicit ``$rest`` placeholder
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Explicitly place the ``$rest`` placeholder in the target URL,
+instead of adding it automatically.
+
+Some times users want to redirect to a different path,
+we have been adding a query parameter in the target URL to
+prevent the old path from being added in the final path.
+For example ``/new/path/?_=``.
+
+Instead of adding the path automatically,
+users have to add the ``$rest`` placeholder in the target URL.
+For example:
+
+From:
+  ``/old/path/$rest``
+To:
+  ``/new/path/$rest``
+
+From:
+  ``/old/path/$rest``
+To:
+  ``/new/path/?page=$rest&foo=bar``
```

Comment:
What about using something like CF here?

```
From: /old/path/*/filename.html
To: /new/:star-1:/filename.html
```

Where, in this example `:star-1:` is whatever `*` matches.

Would that be too complex? If not, would it be useful to be able to have multiple stars? Like `:star-1:`, `:star-2:`, etc...

Reply:
CF allows you to have one single splat/wildcard per redirect. Allowing several wildcards will require us to use regex and iterate over each redirect rule.

Reply:
And netlify allows just one suffix wildcard.

Reply:
OK. We don't probably need that amount of complexity then 😄